site/homepage/static/js/front.js

```diff
@@ -0,0 +1,390 @@
+    window.open($(this).attr('href'))
+  })
+
+  /* animated scrolling */
+  $('.scroll-to, .scroll-to-top').click(function (event) {
+    var fullUrl = this.href
+    var parts = fullUrl.split('#')
+
+    if (parts.length > 1) {
+      scrollTo(fullUrl)
+      event.preventDefault()
+    }
+  })
+
+  function scrollTo (fullUrl) {
+    var parts = fullUrl.split('#')
+    var trgt = parts[1]
+    var targetOffset = $('#' + trgt).offset()
+    var targetTop = targetOffset.top - 100
+
+    if (targetTop < 0) {
+      targetTop = 0
+    }
+
+    $('html, body').animate({
+      scrollTop: targetTop
+    }, 1000)
+  }
+}
+
+/* product detail gallery */
+function productDetailGallery (confDetailSwitch) {
+  $('.thumb:first').addClass('active')
+  var timer = setInterval(autoSwitch, confDetailSwitch)
+
+  $('.thumb').click(function (e) {
+    switchImage($(this))
+    clearInterval(timer)
+    timer = setInterval(autoSwitch, confDetailSwitch)
+    e.preventDefault()
+  })
+
+  $('#mainImage').hover(function () {
+    clearInterval(timer)
+  }, function () {
+    timer = setInterval(autoSwitch, confDetailSwitch)
+  })
+
+  function autoSwitch () {
+    var nextThumb = $('.thumb.active').closest('div').next('div').find('.thumb')
+    if (nextThumb.length === 0) {
+      nextThumb = $('.thumb:first')
+    }
+    switchImage(nextThumb)
+  }
+
+  function switchImage (thumb) {
+    $('.thumb').removeClass('active')
+    var bigUrl = thumb.attr('href')
+    thumb.addClass('active')
+    $('#mainImage img').attr('src', bigUrl)
+  }
+}
+
+/* product detail sizes */
+function productDetailSizes () {
+  $('.sizes a').click(function (e) {
+    e.preventDefault()
+    $('.sizes a').removeClass('active')
+    $('.size-input').prop('checked', false)
+    $(this).addClass('active')
+    $(this).next('input').prop('checked', true)
+  })
+}
+
+$.fn.alignElementsSameHeight = function () {
+  $('.same-height-row').each(function () {
+    var maxHeight = 0
+    var children = $(this).find('.same-height')
+    children.height('auto')
+
+    if ($(window).width() > 768) {
+      children.each(function () {
+        if ($(this).innerHeight() > maxHeight) {
+          maxHeight = $(this).innerHeight()
+        }
+      })
+      children.innerHeight(maxHeight)
+    }
+
+    maxHeight = 0
+    children = $(this).find('.same-height-always')
+    children.height('auto')
+    children.each(function () {
+      if ($(this).height() > maxHeight) {
+        maxHeight = $(this).innerHeight()
+      }
+    })
+    children.innerHeight(maxHeight)
+  })
+}
+
+var windowWidth
+$(function () {
+  windowWidth = $(window).width()
+
+  $(this).alignElementsSameHeight()
+  pictureZoom()
+})
+
+$(window).resize(function () {
+  var newWindowWidth = $(window).width()
+
+  if (windowWidth !== newWindowWidth) {
+    setTimeout(function () {
+      $(this).alignElementsSameHeight()
+      fullScreenContainer()
+      pictureZoom()
+    }, 205)
+    windowWidth = newWindowWidth
+  }
+})
```

Comment:
where does all this code come from?

Reply:
The file is from the theme (Hugo theme "Universal"). It is included to change the configuration of the carousel "Customers". If you want to see what has changed, you unfortunately need to compare the file with the one in the theme. 

The changed part looked before  like the following: 

~~~js
    $('.customers').owlCarousel({
      items: 6,
      itemsDesktopSmall: [990, 4],
      itemsTablet: [768, 2],
      itemsMobile: [480, 1]
    })
~~~

and is now:

~~~js
    $('.customers').owlCarousel({
      items: 4,
      itemsDesktopSmall: [990, 3],
      itemsTablet: [768, 2],
      itemsMobile: [480, 1],
      slideSpeed: 300,
      paginationSpeed: 400,
      autoPlay: true,
      stopOnHover: true,
      singleItem: false
    })
~~~

It configures the [Owl Carousel plugin](http://www.landmarkmlp.com/js-plugin/owl.carousel/index.html#customizing) used by the theme.